java/src/jmri/jmrix/ipocs/configurexml/IpocsConnectionConfigXml.java

```diff
@@ -0,0 +1,85 @@
+package jmri.jmrix.ipocs.configurexml;
+
+import java.io.IOException;
+
+import org.jdom2.Element;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import jmri.configurexml.JmriConfigureXmlException;
+import jmri.jmrix.configurexml.AbstractConnectionConfigXml;
+import jmri.jmrix.ipocs.IpocsConnectionConfig;
+import jmri.jmrix.ipocs.IpocsPortController;
+import jmri.jmrix.ipocs.IpocsSystemConnectionMemo;
+
+/**
+ *
+ * @author Fredrik Elestedt Copyright (C) 2020
+ * @since 4.21.2
+ */
+public class IpocsConnectionConfigXml extends AbstractConnectionConfigXml {
+  private final static Logger log = LoggerFactory.getLogger(IpocsConnectionConfigXml.class);
+  IpocsPortController portController;
+
+  public IpocsConnectionConfigXml() {
+    super();
+  }
+
+  @Override
+  public Element store(Object o) {
+    Element e = new Element("connection");
+    getInstance((IpocsConnectionConfig)o);
+
+    storeCommon(e, portController);
+    e.setAttribute("port", "10000");
+    e.setAttribute("enableAutoConf", "true");
+    e.setAttribute("class", this.getClass().getName());
+
+    extendElement(e);
+    return e;
+  }
+
+  @Override
+  public boolean load(Element shared, Element perNode) throws JmriConfigureXmlException {
+    getInstance();
+    try {
+      // TODO do this
+      //int port = Integer.parseInt(shared.getAttribute("port").getValue());
+      //portController.setPort(port);
+    } catch (NullPointerException ex) {
+    }
+    try {
+      // TODO do this
+      //boolean autoConf = shared.getAttribute("enableAutoConf").getValue().equals("true");
+      //portController.setAutoConf(hostName);
+    } catch (NullPointerException ex) {
+    }
```

Comment:
Is this code that's not finished? Just checking so you haven't forgotten about it.

Reply:
I originally planed to have the port it uses configurable - but just couldn't get it to work... I tried - and failed with the graphical parts of it. Any pointers would be appriciated.
Also, the 2nd option, to disable auto configuration (mDNS) I think can be removed - as the entire IPOCSMR requires it in order to connect.

Reply:
Fixed